server/src/test/java/org/elasticsearch/index/IndexSettingsTests.java

```diff
@@ -339,10 +365,42 @@ public void testStatelessDefaultRefreshInterval() {
                 .put(EXISTING_SHARDS_ALLOCATOR_SETTING.getKey(), "stateless")
                 .build()
         );
-        IndexSettings settings = new IndexSettings(metadata, Settings.EMPTY);
+        IndexSettings settings = new IndexSettings(metadata, Settings.builder().put(STATELESS_ENABLED_SETTING_NAME, true).build());
```

Comment:
This change looks unnecessary? AFAICS, we only need this when setting fast refresh.

We could set it randomly to indicate the independence.

Similar comment for other cases below, i.e., not setting this which the test does not need, except perhaps randomly.

Reply:
I set it because the test was named Stateless, but it truly doesn't check the node setting if it's not fast refresh. I'll remove it simply from the tests that do not need fast refresh. As randomly setting it I think doesn't exercise any code logic.